[Calvin Tan] iP

Cyn7hius opened the pull request:
### DukeLite

> "Human beings in a mob. What's a mob to a ~~king~~ DUKE?" - Frank Ocean (probably)

DukeLite frees your mind of having to remember things you need to do. It's:

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use 

To get started:

1. Download it from [here](https://github.com/Cyn7hius/ip/releases/download/A-Jar/Duke.jar)
2. Click on the icon
3. Add your tasks
4. ???
5. **Profit!** :wink:

Features:

- [x] Manage Tasks
- [x] Find tasks by keyword
- [ ] Add todos, events and deadlines
- [ ] Saved to your device!

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Cyn7hius committed:
Added a basic greeting, and echo everything the user types.

Cyn7hius committed:
Added task list

Cyn7hius committed:
Added ability to mark as done

Cyn7hius committed:
Added new subclasses of Task. Added ability to add 3 different types of tasks: todo, event and deadline

Cyn7hius committed:
Added error handling for wrong commands, improper commands etc

Cyn7hius committed:
Added comments
Added ability to delete tasks

Cyn7hius committed:
Made changes to formatting, handled more edge cases /error inputs

Cyn7hius committed:
Fixed bug where Done and delete only worked for single digit task number

Cyn7hius committed:
Add ability to save task list between each program execution

Cyn7hius committed:
Add ability for deadline tasks to have proper date format

Cyn7hius committed:
Merge branch 'branch-Level-7'

Cyn7hius committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java

Cyn7hius committed:
Refactor code into smaller parts

Cyn7hius committed:
Package the classes into 3 main categories

Cyn7hius committed:
Add Junit tests for tasks

Cyn7hius committed:
Fix Deadline date and time parsing and delete and done not saving to file

Cyn7hius committed:
Add JavaDocs to all classes and most of the methods

Cyn7hius committed:
Format code to follow coding standard

Cyn7hius committed:
Add new command "find" which displays the tasks which contains the keyword

Cyn7hius committed:
Merge branch 'branch-A-JavaDoc'

Cyn7hius committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/Todo.java
#	src/main/java/duke/user/Ui.java

Cyn7hius committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/user/Parser.java
#	src/main/java/duke/user/Ui.java

Cyn7hius committed:
Fix some coding style violations

Cyn7hius committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Cyn7hius committed:
Add ability to build and run Duke using Gradle

Cyn7hius committed:
Add basic GUI

NoraYUuu commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,137 @@
+     * Method which begins the majority of the program execution
+     */
+    public void run() {
+        boolean taskListIsUpdated = false;
```
Maybe can name as isTaskListUpdated for easy identifying boolean? same as below

NoraYUuu commented on src/main/java/duke/user/Parser.java:
```diff
@@ -0,0 +1,196 @@
+     * @return A String[] with each index storing the key information after the input is parsed
+     * @throws DukeException if the command is misused or none of the command words was used
+     */
+    public String[] checkInput(String input) throws DukeException {
```
Might be clearer to name as categorizeInput, since check is quite general.

NoraYUuu commented on src/main/java/duke/user/Ui.java:
```diff
@@ -0,0 +1,105 @@
+        return DIVIDER + "\n" + e.getMessage() + "\n" + DIVIDER;
+    }
+
+    public String displayExceptionMessage(Exception e) {
```
I like how you categorize and name these messages clearly in this class!

NoraYUuu commented on src/main/java/duke/user/Parser.java:
```diff
@@ -0,0 +1,196 @@
+
+    }
+
+    public String checkFind(String input) throws DukeException {
```
I like how you use the noun after check in this class to name and differentiate different check method for it to be easily identified and referred to from other classes.

NoraYUuu reviewed (commented):
Overall the namings are good! Some array names are not in plural form, but after reading the code, it does make sense to use singular form

Cyn7hius committed:
Add custom image to GUI, fix allowing empty description for tasks

Cyn7hius committed:
Merge branch 'branch-Level-10'

Cyn7hius committed:
Add Checkstyle

Cyn7hius committed:
Use Checkstyle to fix formatting. Change Gradle to include Checkstyle when building

Cyn7hius committed:
Merge branch 'branch-A-Checkstyle'

CSYGalvin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,149 @@
+                output += ui.displayDoneTaskMessage(current.toString());
+                taskListIsUpdated = true;
+
+                // case where user wants to delete a task item, similar to done
```
Good job on writing comments to make your code easier to read. Should you capitalize `case` so that it is consistent with the other comments?

CSYGalvin commented on src/main/java/duke/data/TaskList.java:
```diff
@@ -0,0 +1,83 @@
+    /**
+     * Gets a specific task from the taskList.
+     *
+     * @param index The index of the item in the taskList (0-indexed)
```
Looks clean! Just missed a punctuation mark here.

CSYGalvin commented on src/main/java/duke/data/TaskList.java:
```diff
@@ -0,0 +1,83 @@
+    /**
+     * Finds the task that contains the keyword.
+     *
+     * @param keyword the word to be included in the task description
```
In some javadoc comment, you capitalize the parameter description while others you do not. Maybe you can make it consistent across the project?

CSYGalvin commented on src/main/java/duke/user/Parser.java:
```diff
@@ -0,0 +1,224 @@
+    }
+
+    /**
+     * Axillary function to check that the done command word is used correctly.
```
Maybe you can elaborate what you mean by `used correctly`?

CSYGalvin reviewed (commented):
Good job!! I like how you included many comments showing how your project works

Cyn7hius committed:
Change mainClass in build-gradle

Cyn7hius committed:
Add assert statements

These statements allows me to verify that there is a proper taskList and Ui object initialised.
It allows me to better debug and identify the fault if any of the assert statements fail

Cyn7hius committed:
Clean code following textbook's Code Quality standard

Made changes to my code quality including:
1. Removing magic literals
2. using as little nested operations as possible
3. Adding comments where appropriate
4. Removing duplicate code and merging them together

Cyn7hius committed:
Merge pull request #1 from Cyn7hius/branch-A-Assertions

Add assert statements

Cyn7hius committed:
Merge branch 'master' into branch-A-CodeQuality

Cyn7hius committed:
Merge pull request #2 from Cyn7hius/branch-A-CodeQuality

Branch a code quality

Cyn7hius committed:
Add a help button which opens a new window displaying the command list

The command menu is useful as a quick reference to all the available commands available in Duke.
Moreover as a separate window, it allows the user to refer to the command list at the side while
using Duke.

Cyn7hius committed:
Merge branch 'branch-C-Help'

Li-Lehao commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,149 @@
+
+        Parser parser = new Parser(tasks);
+        boolean taskListIsUpdated = false;
+        boolean taskListIsAddedTo = false;
```
For boolean variables it is better to start with is or has. Maybe we can consider changing this name to hasAddedTaskList? I have no difficulty understanding your naming, but maybe it is better to start with is or has.

Li-Lehao commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,149 @@
+    /**
+     * Handles the user input and sends it to the parser to perform the commands, if valid.
+     */
+    public String getResponse(String input) {
```
I really like the way you implement this getResponse() method. It is very neat and easy to understand.

Li-Lehao commented on src/main/java/duke/data/Storage.java:
```diff
@@ -0,0 +1,110 @@
+                int taskDescIndex = 2;
+                int dueDateTimeIndex = 3;
+
+                if (taskDetails.startsWith("D ")) {
+                    taskList.add(new Deadline(taskDetails.split(Ui.SPLIT_DELIMITER)[taskDescIndex],
+                            LocalDate.parse(taskDetails.split(Ui.SPLIT_DELIMITER)[dueDateTimeIndex].split(" ")[0]),
+                            LocalTime.parse(taskDetails.split(Ui.SPLIT_DELIMITER)[dueDateTimeIndex].split(" ")[1]),
+                            !taskDetails.split(Ui.SPLIT_DELIMITER)[isDoneFlagIndex].equals("0")));
+
+                } else if (taskDetails.startsWith("E ")) {
+                    taskList.add(new Event(taskDetails.split(Ui.SPLIT_DELIMITER)[taskDescIndex],
+                            taskDetails.split(Ui.SPLIT_DELIMITER)[dueDateTimeIndex],
+                            !taskDetails.split(Ui.SPLIT_DELIMITER)[isDoneFlagIndex].equals("0")));
+
+                } else if (taskDetails.startsWith("T ")) {
+                    taskList.add(new Todo(taskDetails.split(Ui.SPLIT_DELIMITER)[taskDescIndex],
+                            !taskDetails.split(Ui.SPLIT_DELIMITER)[isDoneFlagIndex].equals("0")));
```
Although you separated the lines, this part of code still seems a little bit hard to understand. Maybe we can consider separate them more.

Li-Lehao commented on src/main/java/duke/data/TaskList.java:
```diff
@@ -0,0 +1,83 @@
+     *
+     * @return the number of tasks in the taskList
+     */
+    public int getLength() {
```
getAmountOfTasks() or getSize() maybe more meaningful. 
Overall you did a really good job. Your code is very neat and elegant with very few violations of naming rules.

Cyn7hius committed:
Add background for GUI, Add README and screenshots

Cyn7hius committed:
Set theme jekyll-theme-slate

Cyn7hius committed:
Change Duke to Dino

Cyn7hius committed:
Merge branch 'master' of https://github.com/Cyn7hius/ip

Cyn7hius committed:
Update GUI image for website

Cyn7hius committed:
Update config files

Cyn7hius committed:
Update Dino icon, update README

Cyn7hius committed:
Clean up code. Update README.

Applied SLAP and extracted long methods in Dino.java and Parser.java into much smaller sub methods.
This reduced the method length by more than half in both methods.

Deleted unnecessary comments, classes and methods which are not used anymore.

Ran checkStyle and checked that all the code followed the style guide recommended.

Cyn7hius committed:
Update GUI screenshot
